.ci/pitest.groovy

```diff
@@ -0,0 +1,459 @@
+        final String exceptionMessage = "\nUnexpected argument: ${argument}" + usageString
+        throw new IllegalArgumentException(exceptionMessage)
+    }
+    return exitCode
+}
+
+/**
+ * Parse the pom.xml file to get all the available pitest profiles.
+ *
+ * @return A set of all available pitest profiles
+ */
+private static Set<String> getPitestProfiles() {
+    final GPathResult mainNode = new XmlSlurper().parse(".${SEPARATOR}pom.xml")
+    final NodeChildren ids = mainNode.profiles.profile.id as NodeChildren
+    final Set<String> profiles = new HashSet<>()
+    ids.each { node ->
+        final GPathResult id = node as GPathResult
+        final String idText = id.text()
+        if (idText.startsWith("pitest-")) {
+            profiles.add(idText)
+        }
+    }
+    return profiles
+}
+
+/**
+ * Generate the pitest report by running the respective maven command.
+ *
+ * @param profile the pitest profile to execute
+ */
+private static void generatePitestReport(String profile) {
+    final String mavenCommand = 'mvn --no-transfer-progress -e -P"' +
+            profile + '" clean test-compile org.pitest:pitest-maven:mutationCoverage'
+    final Process proc = mavenCommand.execute()
+    proc.in.eachLine { println it }
+}
+
+/**
+ * Check the generated pitest report. Parse the survived and suppressed mutations and compare
+ * them.
+ *
+ * @param profile the pitest profile to execute
+ * @param flag command line argument flag to determine output format
+ * @return {@code 0} if pitest report is as expected, {@code 1} otherwise
+ */
+private static int checkPitestReport(String profile, String flag) {
+    final XmlParser xmlParser = new XmlParser()
+    File mutationReportFile = null
+    final String suppressedMutationFileUri =
+            ".${SEPARATOR}.ci${SEPARATOR}pitest-suppressions${SEPARATOR}${profile}-suppressions.xml"
+    new File(".${SEPARATOR}target${SEPARATOR}pit-reports")
+            .eachFileRecurse(FileType.FILES) {
+                if (it.name == 'mutations.xml') {
+                    mutationReportFile = it
+                }
+            }
+    final Node mutationReportNode = xmlParser.parse(mutationReportFile)
+    final Set<Mutation> survivedMutations = getSurvivedMutationsList(mutationReportNode)
+
+    final File suppressionFile = new File(suppressedMutationFileUri)
+    Set<Mutation> suppressedMutations = Collections.<Mutation> emptySet()
+    if (suppressionFile.exists()) {
+        final Node suppressedMutationNode = xmlParser.parse(suppressedMutationFileUri)
+        suppressedMutations = getSuppressedMutationList(suppressedMutationNode)
+    }
+    return compareMutations(survivedMutations, suppressedMutations, flag)
+}
+
+/**
+ * Get the survived mutation list. All child nodes of the main {@code mutations} node
+ * are parsed.
+ *
+ * @param mainNode the main {@code mutations} node
+ * @return A set of survived mutations
+ */
+private static Set<Mutation> getSurvivedMutationsList(Node mainNode) {
```

Comment:
`getSurvivedMutationsList` -> `getSurvivingMutationList` to be consistent with method naming (no need for plural)

Reply:
Done.